nav2_collision_monitor/src/pointcloud.cpp

```diff
@@ -76,16 +77,29 @@ void PointCloud::getData(
     return;
   }
 
-  // Obtaining the transform to get data from source frame and time where it was received
-  // to the base frame and current time
+
   tf2::Transform tf_transform;
-  if (
-    !nav2_util::getTransform(
-      data_->header.frame_id, data_->header.stamp,
-      base_frame_id_, curr_time, global_frame_id_,
-      transform_tolerance_, tf_buffer_, tf_transform))
-  {
-    return;
+  if (base_shift_correction_) {
+    // Obtaining the transform to get data from source frame and time where it was received
+    // to the base frame and current time
+    if (
+      !nav2_util::getTransform(
+        data_->header.frame_id, data_->header.stamp,
+        base_frame_id_, curr_time, global_frame_id_,
+        transform_tolerance_, tf_buffer_, tf_transform))
+    {
+      return;
+    }
+  } else {
+    // Obtaining the transform to get data from source frame to base frame without time shift
+    // considered. Less accurate but much more faster option.
```

Comment:
```suggestion
    // considered. Less accurate but much more faster option not dependent on state estimation frames.
```

Ditto in other locations - this is mostly a note for myself in the future

Reply:
Done in https://github.com/ros-planning/navigation2/pull/3412/commits/7e432ecf90f8982cf38f88a527406becf5e9090c